nexus/test-utils/src/lib.rs

```diff
@@ -23,6 +23,9 @@ use std::path::Path;
 use std::time::Duration;
 use uuid::Uuid;
 
+// Expose the version of Nexus used by the test utils.
+pub use omicron_nexus as nexus;
```

Comment:
If the caller depended on omicron_nexus themselves, is there any real chance it could differ (since it would be a path dependency in the same repo)?

Not saying this is a bad pattern... now I wonder: should we do the same thing with sled agent instead of having omicron-nexus have a dev-dep on omicron-sled-agent?

Reply:
This is related to my comment here: https://github.com/oxidecomputer/omicron/pull/1835#discussion_r1001254054

- `nexus-test-utils`, under `nexus/test-utils`, has a weird cyclic relationship with the `omicron-nexus` crate (note: It has this relationship independent of this PR). `nexus-test-utils` depends on `omicron-nexus`, but `omicron-nexus` has a `dev-dependency` back on `nexus-test-utils`.
- These both *appear* to be path-based dependencies. 
- If I *don't* use this version of nexus, and instead, in the `disk_create.rs` test, I attempt to use Nexus symbols directly, I get errors like the following:

```
error[E0308]: mismatched types                                                                                         
   --> nexus/src/app/sagas/disk_create.rs:662:38                                                                       
    |                                                                                                                  
662 |         let params = new_test_params(&opctx, project_id);                                                        
    |                      --------------- ^^^^^^ expected struct `context::OpContext`, found struct `omicron_nexus::context::OpContext`
    |                      |                                                                                                                                                                                                                   
    |                      arguments to this function are incorrect                                                                                                                                                                            
    |                                                                                                                                                                                                                                          
    = note: struct `omicron_nexus::context::OpContext` and struct `context::OpContext` have similar names, but are actually distinct types
note: struct `omicron_nexus::context::OpContext` is defined in crate `omicron_nexus`                                                                                                                                                           
   --> /home/smklein/repos/oxide/omicron/nexus/src/context.rs:223:1                                                    
    |                                                                                                                  
223 | pub struct OpContext {                                                                                           
    | ^^^^^^^^^^^^^^^^^^^^                                                                                             
note: struct `context::OpContext` is defined in the current crate                                                      
   --> nexus/src/context.rs:223:1                                                                                      
    |                                                                                                                  
223 | pub struct OpContext {                                                                                           
    | ^^^^^^^^^^^^^^^^^^^^                                                                                             
note: function defined here                                                                                            
   --> nexus/src/app/sagas/disk_create.rs:633:8                                                                        
    |                                                                                                                  
633 |     fn new_test_params(opctx: &OpContext, project_id: Uuid) -> Params {                                          
    |        ^^^^^^^^^^^^^^^ -----------------                                                 
```

So, for whatever reason, they can't be consolidated as a single type.

This doesn't *seem* necessary for accessing the `omicron-sled-agent` dependency, for whatever reason. I'm hesitant to start exposing all the internals of `nexus-test-utils` publicly though -- for any crate used by a Nexus test, that *also* happens to be in `nexus-test-utils`, should I re-export it? There's a lot more than just the sled agent here; nexus-test-utils and nexus both have dependencies on:

- anyhow
- chrono
- dropshot
- headers
- http
- omicron-common
- oximeter
- oximeter-client
- oximeter-producer
- serde
- serde_json
- serde_urlencoded
- slog
- uuid

Should I also be re-exporting all these crates?

Reply:
It definitely doesn't make sense to me to expose all of those.  But does it not feel strange that we have to do this with just this one crate?  This sort-of-cyclic relationship seems kind of fraught but I don't have any specific suggestion for improving it so :shrug:

Reply:
See the explanation above, surrounding e44438d85bd30d82fff99bd9b30b156289afef64 . This circular dependency has been... not entirely removed, but reduced enough that I don't need this `pub use` statement.